odoo/fields.py

```diff
@@ -112,14 +112,15 @@ class Field(MetaField('DummyField', (object,), {})):
 
     :param bool required: whether the value of the field is required (default: ``False``)
 
-    :param bool index: whether the field is indexed in database. Note: no effect
-        on non-stored and virtual fields. (default: ``False``) Possible attributes:
+    :param str index: whether the field is indexed in database, and the kind of index.
+        Note: this has no effect on non-stored and virtual fields.
+        The possible values are:
 
-        * ``True``: auto-detect (currently set to btree)
-        * ``btree``: standard index, good for many2one
-        * ``not null``: btree without indexing NULL values (usefull when 90% of values are NULL)
-        * ``gin``: Generalized Inverted Indexes (GIN). Better for full-text search.
-        * ``False``: no index (default)
+        * ``"btree"`` or ``True``: standard index, good for many2one
+        * ``"btree_not_null"``: same as ``"btree"``, but without indexing NULL values
+                                (useful when 90% of values are NULL)
```

Comment:
:suspect: This random stat is just confusing . It still "useful" at 50 %.

Reply:
Well it is only ever useful if you don't do NULL searches which is a pretty critical piece of information to omit. That's why I questioned some of the choices on original PR, because the more common search is NULL's for those fields .

Reply:
> Well it is only ever useful if you don't do NULL searches which is a pretty critical piece of information to omit

Yeah, but it is still not completely correct: 
If your column is sparse (50% by example), a search with IS NULL it won't be faster with the index than without  (extra cost of reading the index vs seq scan)

Reply:
@ryv-odoo yes you are right, and I just learned something, but the argument to not index NULL's is to save space, it doesn't make the scan any faster for the NOT NULL cases.  And the decisions were based of what looked like a non stock trading company. I think you need to be a bit judicious about whether it makes sense for all use cases, especially when stock is involved. 

In any case, I just tested on a variety of tables with lots of records. The planner at least in pg13 is more than capable of working out when it makes sense to use the index for a NULL search and when to seq scan.

Reply:
> I think you need to be a bit judicious about whether it makes sense for all use cases

Yes, agree. Like @rco-odoo said, it was clearly rushed. In some case, it won't be an easy tradeoff.
Thank you for the feedback.